pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -204,16 +206,8 @@ func convertServices(cfg config.Config, clusterID cluster.ID) []*model.Service {
 	hostAddresses := []*HostAddress{}
 	for _, hostname := range serviceEntry.Hosts {
 		localAddresses := addresses
-		if len(localAddresses) == 0 {
-			// we have no user-assed addresses but we can check if we have auto-assigned addresses
-			if autoAddresses, ok := addressLookup[hostname]; ok {
-				for _, aa := range autoAddresses {
-					localAddresses = append(localAddresses, aa.String())
-				}
-			}
-		}
```

Comment:
@howardjohn, I removed this bit where we add auto-assigned addresses to our list of addresses so if we do use auto-assigned addresses it means that our list of user-assigned addresses is 0 length which means the hostAddress should get a constants.UnspecifiedIP and then not set ClusterVIPs.